### PR TITLE
#162727482 Implement report art feature

### PR DESCRIPTION
**Pull request by abejide001**
#### What does this PR do?
Enable users to report an art they find offensive
#### Description of Task to be completed?
- Implement the model  and migrations for report
- Create associations between the report, user model, and report, art model
- Implement validation for the inputs
- Implement create a report method
- Implement get all reports method
- Write test for the report controller
#### How should this be manually tested?
- Run `npm i`
- Created db `createdb dbname`
- Run the migrations `npm run db:migrate`
- Run the seeders `npm run db:seed`
- Start the server `npm run start:dev`
- Visit the endpoint to create a report `localhost:9000/api/v1/arts/:id/report`, enter your token and the following keys `reportType` and `reportText`
#### Any background context you want to provide?
- The report type consists of `Spam, Abuse, Indecency`
#### What are the relevant pivotal tracker stories?
[#162727482 ](https://www.pivotaltracker.com/story/show/162727482)
#### Screenshots (if appropriate)
screen shot 2019-01-23 at 10 19 58 am

### CODE REVIEW

File: src/controllers/ReportController.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+   * @param {object} req
+   * @param {object} res
+   * @memberof ReportsController
+   * @returns {Object} report type and text
+   */
+  static async createReport(req, res) {
+    try {
+      const { reportText } = req.body;
+      const artId = await Art.findById(req.params.id);
+      if (!artId) {
```
Comment (andela-dbamidele) [inline src/controllers/ReportController.js]: Please move this to the middleware.

File: src/controllers/ReportController.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  /**
+   * @desc GET /api/v1/arts/reports
+   * @param {object} req
+   * @param {object} res
+   * @memberof ReportsController
+   * @returns {Object} report type and text
+   */
+  static async getAllReports(req, res) {
+    console.log('aaaaaaaa')
```
Comment (andela-dbamidele) [inline src/controllers/ReportController.js]: Please remove console statement.

**Review by andela-dbamidele — CHANGES_REQUESTED** [review]
Please make the changes as soon as possible.

File: src/controllers/ReportController.js (hunk 1 above)
Comment (abejide001) [reply src/controllers/ReportController.js]: okay, fixed

File: src/controllers/ReportController.js (hunk 2 above)
Comment (abejide001) [reply src/controllers/ReportController.js]: okay

**Comment by abejide001** [comment]
> Please make the changes as soon as possible.

fixed

**Pull request merged**